### ISSUE TITLE
Multiple WebView2 instances hosted in the same window crashed at once

### ISSUE THREAD

**Issue by szanto90balazs** [MicrosoftEdge/WebView2Feedback#3202]
**Description**
Multiple WebView2 instances (tabs) hosted in the same window crashed (reported by CoreWebView2.ProcessFailed) and this is what we've found looking at the crash dump:

```
000000bb`b0bfd090 00007ff8`61fdac6f     : 000000bb`b0bfd190 000000bb`b0bfd198 00000000`00000000 00000000`00000000 : msedge!content::WebContentsImpl::GetResponsibleWebContents+0xb
000000bb`b0bfd0c0 00007ff8`62005422     : 00000000`00000000 000062f0`09610280 aaaaaaaa`aaaaaa01 000062f0`0ccd4f00 : msedge!performance_manager::GetWebContentsTopWindow+0x2f
000000bb`b0bfd0f0 00007ff8`5f85a1e4     : aaaaaaaa`aaaaaaaa 000000bb`b0bfd230 000062f0`0c558bb0 00007ff8`600b4ebc : msedge!performance_manager::PerformanceManagerTabHelper::OnVisibilityChanged+0x42
000000bb`b0bfd180 00007ff8`615064f2     : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : msedge!content::WebContentsImpl::SetVisibilityAndNotifyObservers+0x230
000000bb`b0bfd2c0 00007ff8`620e7826     : 000062f0`0cb853a0 00000000`00000024 80000000`00000028 aaaaaaaa`aaaaaaaa : msedge!content::WebContentsImpl::UpdateVisibilityAndNotifyPageAndView+0x2de
000000bb`b0bfd400 00007ff8`5fc8376f     : 000062f0`00044c00 00000000`00000000 000000bb`b0bfd6c0 000000bb`b0bfd4f0 : msedge!content::WebContentsImpl::OnRenderFrameProxyVisibilityChanged+0x106
000000bb`b0bfd4a0 00007ff8`5fc82c64     : 000043bd`e031a10b 00000000`0c510901 000043bd`e031a0ab 000062f0`0a487000 : msedge!content::CrossProcessFrameConnector::OnVisibilityChanged+0x6f
000000bb`b0bfd4e0 00007ff8`5e820be5     : 00000000`44892000 00000000`00000000 000062f0`0a5cd000 00000000`00000000 : msedge!blink::mojom::RemoteFrameHostStubDispatch::Accept+0x42e
000000bb`b0bfd8c0 00007ff8`5e81f4a4     : aaaaaaaa`aaaaaaaa aaaaaaaa`aaaaaaaa aaaaaaaa`aaaaaaaa aaaaaaaa`aaaaaaaa : msedge!mojo::InterfaceEndpointClient::HandleIncomingMessageThunk::Accept+0xcf5
000000bb`b0bfdac0 00007ff8`5f7e362b     : aaaaaaaa`aaaaaaaa aaaaaaaa`aaaaaaaa aaaaaaaa`aaaaaaaa aaaaaaaa`aaaaaaaa : msedge!mojo::InterfaceEndpointClient::HandleIncomingMessage+0x33a
000000bb`b0bfdcd0 00007ff8`5e3248ef     : aaaaaaaa`00000000 00000001`00000038 000043bd`e031ad1b 00000000`00000000 : msedge!base::internal::Invoker<base::internal::BindState<void (__cdecl IPC::`anonymous namespace'::ChannelAssociatedGroupController::*)(mojo::Message) __ptr64,scoped_refptr<IPC::`anonymous namespace'::ChannelAssociatedGroupController>,Message::IPC>,void __cdecl(void)>::RunOnce+0x37b
000000bb`b0bfdff0 00007ff8`5e3bffd3     : 00000000`00000000 00006d94`00223050 00006d94`0028c280 00006d94`0028c280 : msedge!base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork+0x178f
000000bb`b0bfe3b0 00007ff8`600bf5e1     : 00000000`00000000 00007ff8`6a4dad18 000043bd`e03197bb 00006d94`00222f98 : msedge!base::MessagePumpForUI::DoRunLoop+0x1e3
000000bb`b0bfe550 00007ff8`5fc0572d     : 00000000`00000008 000000bb`b0bfe6a0 000000bb`b0bfe6a8 00007ff8`6005023f : msedge!base::MessagePumpWin::Run+0x71
000000bb`b0bfe5b0 00007ff8`608c2cc7     : 000043bd`e031977b 00000000`00000008 00000000`00000000 00007ff8`6a4dad18 : msedge!base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run+0xdd
000000bb`b0bfe640 00007ff8`605c31f0     : 00000000`00000000 00000000`00000000 00000000`00000000 00000088`d0aaa15c : msedge!base::RunLoop::Run+0x117
000000bb`b0bfe780 00007ff8`605c3000     : 00000000`00000001 000000bb`b0bfe8e0 000000bb`b0bfe8f8 000000bb`b0bfe920 : msedge!content::BrowserMainLoop::RunMainMessageLoop+0x9a
000000bb`b0bfe7f0 00007ff8`605c2d52     : 000000bb`b0bfe9c8 00007ff8`6a4dad18 00000000`00000018 00000000`00000000 : msedge!content::BrowserMain+0xe0
000000bb`b0bfe890 00007ff8`605c22ab     : 00007ff8`69f0e91f 000000bb`b0bfea10 00007ff9`0c095e90 00006d94`002500f0 : msedge!content::RunBrowserProcessMain+0xd2
000000bb`b0bfe990 00007ff8`60585ebb     : 00006d94`00250410 00000000`0000000d 000000bb`b0bfeb40 00006d94`00250120 : msedge!content::ContentMainRunnerImpl::RunBrowser+0x45d
000000bb`b0bfeae0 00007ff8`6058507f     : 00006d94`002500f0 00007ff8`6b4b4a88 000043bd`e031995b 00006d94`002640d0 : msedge!content::ContentMainRunnerImpl::Run+0x2fb
000000bb`b0bfec30 00007ff8`60583cae     : 00000000`b0bfef00 00006d94`00244600 000000bb`b0bff090 00000267`20c04980 : msedge!content::ContentMain+0x21a
000000bb`b0bfee50 00007ff7`b1332fce     : 00000000`b0bff000 00007ff8`60583a30 0000028c`00290600 0000028c`002781b0 : msedge!ChromeMain+0x27e
000000bb`b0bff030 00007ff7`b12c6b8e     : 00000000`00000000 aaaaaaaa`aaaaaaaa 00000000`00000000 0000028c`0023c280 : msedgewebview2!MainDllLoader::Launch+0x408
000000bb`b0bff2d0 00007ff7`b12f40f2     : 00000000`00000000 00007ff7`b12f4169 00000000`00000000 00000000`00000000 : msedgewebview2!wWinMain+0x521
000000bb`b0bff800 00007ff9`0c0926bd     : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : msedgewebview2!__scrt_common_main_seh+0x106
000000bb`b0bff840 00007ff9`0d5adfb8     : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : kernel32!BaseThreadInitThunk+0x1d
000000bb`b0bff870 00000000`00000000     : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : ntdll!RtlUserThreadStart+0x28
```

**Version**
SDK: 1.0.1418.22
Runtime: 107.0.1418.26
Framework: WPF
OS: Win10

**Repro Steps**

No repro steps found unfortunately.

[AB#43461975](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/43461975)

**Comment by szanto90balazs** [comment]
No repro steps found unfortunately, but I do wonder if you already track a crash like this internally.

**Comment by ElyssaJyu** [comment]
Hi @szanto90balazs, could you give some detailed description about your workaround? Could you get a trace? Thanks.

**Comment by ElyssaJyu** [comment]
I have simply created a WPF program with two webview2 in the same window, and it works. If you could provide some sample about your scenario, it will help to resolve the issue. Thanks.

**Comment by szanto90balazs** [comment]
Hey @ElyssaJyu, thanks for getting back to me. Unfortunately, I can't reproduce the issue, it only happened few times, but I was interested in if you already ran into the same stack trace provided and I was hoping you already track something like this.

@ElyssaJyu Would sending the crash dump file itself help diagnosing or narrowing down the issue? As it contains sensitive information, I didn't want to upload it here, but I can share it privately.

**Comment by ElyssaJyu** [comment]
Hey @szanto90balazs, the root cause of crash can be various, so I might not be able to ensure whether we have already tracked the same or similar trace. Multiple webview2 controls is expected to enable host in one window. Could you give a description about how you use this scenario? Also, if you are willing to share it privately, you can send it to sunelyssa@microsoft.com. Thanks.

**Comment by szanto90balazs** [comment]
@ElyssaJyu Our scenarios is about hosting multiple WebView2 instances per window in a tabbed manner. I sent you an e-mail including the dump. Thanks in advance 👍

**Comment by ElyssaJyu** [comment]
Hi @szanto90balazs, Thaks for your provided information, I've added this to our backlog to take a look.

**Comment by szanto90balazs** [comment]
@ElyssaJyu Thanks for adding it to your backlog. Did the dump turn out to be useful? Any ETA when this is going to be picked up and addressed? Thanks 👍

**Comment by ElyssaJyu** [comment]
Hi @szanto90balazs, the dump does helps but not that much. Does this crash happen again currently? If you could provide more details (i.e. ETA), it will be helpful. And I think I saw a similar crash has been tracked internally. I will sync with you when any update. Thanks.

**Comment by ElyssaJyu** [comment]
Also, I saw the similar crash you post before, does [this repro](https://github.com/MicrosoftEdge/WebView2Feedback/issues/2359#issuecomment-1144834587) still cause the crash, or this post is another scenario to trigger the crash? Could you describe the scenario that this crash issue can be triggered?

**Comment by szanto90balazs** [comment]
Hey @ElyssaJyu, it doesn't happen currently and we saw it happening few times, but lately not ran into it. Also the other repro you linked doesn't cause any crash at the moment.

You mentioned sawing a similar crash tracked internally, can you give some details on potential repro steps or some leads, so I can also start trying to break our app or create a sample app which may or may not reproduce the issue. Thanks 👍

**Comment by ElyssaJyu** [comment]
Hey @szanto90balazs, I'm still investigating on the cause of this crash and potential repro steps. There are not many repro details here, though there is a similar crash internally. I will sync with you when updated. Sorry for that inconvenience.

**Comment by LiangTheDev** [comment]
@szanto90balazs there is a fix made in Edge 112 for this. Could you please check whether the issue has been resolved with latest version of Edge WebView2 Runtime?

**Comment by champnic** [comment]
This is fixed in runtimes 112.0.1716.0+. Thanks!
